File: Java/Language/Basics/src/Banker.java
package banking;

//factory class
public class Banker {

	private static long nid = 1;

	public static Account openCurrentAccount() {
		var acc = new CurrentAccount();
		acc.id = 1000000 + nid++;
		return acc;
	}

	public static Account openSavingsAccount() {
		var acc = new SavingsAccount();
		acc.id = 2000000 + nid++;
		return acc;
	}

	//all members of this class are static so there's no reason to create its instances
	private Banker() {}
}

File: CSharp/Basics/InterfaceTest1/Banking/Banker.cs
namespace Banking
{
	public static class Banker
	{
		private static long nid = System.DateTime.Now.Ticks % 1000000;

		public static Account OpenCurrentAccount()
		{
			return new CurrentAccount()
			{
				Id = 100000000L + nid++
			};
		}

		public static Account OpenSavingsAccount()
		{
			return new SavingsAccount()
			{
				Id = 200000000L + nid++
			};
		}
	}
}

